lib/ruby_marks/default/find_marks.rb

```diff
@@ -0,0 +1,244 @@
+          block[:x1] <= x && block[:x2] >= x &&
+            block[:y1] <= y && block[:y2] >= y
+        end
+      end
+
+      def find_map_lines
+        line = 0
+        mark_ant = nil
+        blocks.sort! { |a, b| a[:y1] <=> b[:y1] }
+
+        blocks.each do |mark|
+          next unless mark_is_out_of_group?(mark, group)
+
+          mark_positions = mark[:y1] - 10..mark[:y1] + 10
+          line += 1 unless mark_ant && mark_positions.include?(mark_ant[:y1])
+          mark[:line] = line
+          mark_ant = mark
+        end
+        
+        # remove marks not mapped
+        blocks.delete_if { |m| m[:line].nil? }
+        # sort by lines
+        blocks.sort_by! { |a| [a[:line], a[:x1]] }
+      end
+
+      def mark_is_out_of_group?(mark, group)
+        mark_width  = ImageUtils.calc_width(mark[:x1], mark[:x2])
+        mark_height = ImageUtils.calc_height(mark[:y1], mark[:y2])
+
+        mark_width >= group.mark_width_with_down_tolerance &&
+                      mark_height >= group.mark_height_with_down_tolerance
+      end
+
+      def delete_conflicts
+        mark_ant = nil
+        blocks.each do |mark|
+          if mark_ant && mark_ant[:line] == mark[:line]
+            mark_ant_center = ImageUtils.image_center(mark_ant)
+            mark_center     = ImageUtils.image_center(mark)
+            if (mark_ant_center[:x] - mark_center[:x]).abs < 10
+              mark[:conflict] = true
+              mark[:conflicting_mark] = mark_ant
+            else
+              mark_ant = mark
+            end
+          else
+            mark_ant = mark
+          end
+        end
+        blocks.delete_if { |m| m[:conflict] }
+      end
+
+      def verify_bubbles_adjust
+        first_position = 0
+        elements_position_count = 0
+        blocks.map { |m| m[:line] }.each do |dash|
+          marks = blocks.select { |m| m[:line] == dash }
+          if marks.count == group.marks_options.count
+            first_position += marks.first[:x1]
+            elements_position_count += 1
+          end
+        end
+
+        if elements_position_count.positive?
+          first_position /= elements_position_count
+          distance = group.distance_between_marks * (group.marks_options.count - 1)
+          last_position = first_position + distance
+          blocks.delete_if do |mark|
+            mark[:x1] < first_position - 10 ||
+              mark[:x1] > last_position + 10
+          end
+
+          blocks.map { |m| m[:line] }.each do |dash|
+            loop do
+              reprocess = false
+              marks = blocks.select { |m| m[:line] == dash }
+              marks.each_with_index do |current_mark, index|
+                if index.zero?
+                  first_mark_position = first_position - 5..first_position + 5
+                  unless first_mark_position.include?(current_mark[:x1])
+                    new_mark = { x1: first_position,
+                                x2: first_position + group.mark_width,
```

Comment:
Can you please fix the indentation of this block?

Reply:
I'll fix it